compiler-core/src/ast.rs

```diff
@@ -377,15 +377,16 @@ pub struct Import<PackageName> {
     pub documentation: Option<SmolStr>,
     pub location: SrcSpan,
     pub module: SmolStr,
-    pub as_name: Option<SmolStr>,
     pub unqualified: Vec<UnqualifiedImport>,
     pub package: PackageName,
+    pub alias: Option<(AssignName, SrcSpan)>,
 }
 impl<T> Import<T> {
     pub(crate) fn used_name(&self) -> SmolStr {
-        self.as_name
+        self.alias
             .as_ref()
             .cloned()
+            .map(|(name, _)| name.name().into())
```

Comment:
When the module is discarded this will result in it being treated as if it was aliased. We'd want the variable name only, and the method will have to return none when it is discarded.

Probably will need a test in the language server bit to ensure this is correct.